packages/core/src/BaseNavigationContainer.tsx

```diff
@@ -122,20 +125,43 @@ export const BaseNavigationContainer = React.forwardRef(
 
     const { keyedListeners, addKeyedListener } = useKeyedChildListeners();
 
-    const dispatch = useLatestCallback(
-      (
-        action:
-          | NavigationAction
-          | ((state: NavigationState) => NavigationAction)
-      ) => {
+    const pendingActionsRef = React.useRef<NavigationDispatchAction[]>([]);
+    const isReadyRef = React.useRef(false);
+
+    const dispatchAction = useLatestCallback(
+      (action: NavigationDispatchAction) => {
         if (listeners.focus[0] == null) {
           console.error(NOT_INITIALIZED_ERROR);
+          return false;
         } else {
           listeners.focus[0]((navigation) => navigation.dispatch(action));
+          return true;
         }
       }
     );
 
+    const dispatch = useLatestCallback((action: NavigationDispatchAction) => {
+      if (listeners.focus[0] == null) {
+        pendingActionsRef.current.push(action);
+        console.error(NOT_INITIALIZED_ERROR);
+        return;
+      }
+
+      if (!isReadyRef.current) {
```

Comment:
Not sure I understand what this is doing. `isReadyRef` is based on `listeners.focus[0]`, but it's printing an error in one case, and a warning in other, and both cases it gets queued either way.